### ISSUE TITLE
Cannot start tower job with ansible

### ISSUE THREAD

**Issue by dforste** [ansible/tower-cli#666]
I am unable to run a tower job. The inventory doesn't work with null or with the inventory ID or Inventory name. 

Versions: 
```
Tower 3.4.1 

Tower CLI 3.3.2

ansible 2.7.5
  config file = /root/ansible/os-updates/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Sep 12 2018, 05:31:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

Ansible task: 
```
- name: Run tower jboss stop job.
  delegate_to: localhost
  tower_job_launch:
    limit: "{{ inventory_hostname }}"
    inventory: "2"
    tower_host: https://tower.example.com
    tower_username: "{{ secret_tower_user }}"
    tower_password: "{{ secret_tower_password }}"
    job_template: "Jboss restart"
    tags: stop
  register: job
```

Error 
```
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1550596043.35-82061054507257 `" && echo ansible-tmp-1550596043.35-82061054507257="` echo /root/.ansible/tmp/ansible-tmp-1550596043.35-82061054507257 `" ) && sleep 0'
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/ansible_tower.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/basic.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/parsing/convert_bool.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/_collections_compat.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/six/__init__.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/_text.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/parsing/__init__.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/pycompat24.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/process.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/__init__.py
Using module_utils file /usr/lib/python2.7/site-packages/ansible/module_utils/common/file.py
Using module file /usr/lib/python2.7/site-packages/ansible/modules/web_infrastructure/ansible_tower/tower_job_launch.py
<localhost> PUT /root/.ansible/tmp/ansible-local-17120fM55LR/tmp1DLY5_ TO /root/.ansible/tmp/ansible-tmp-1550596043.35-82061054507257/AnsiballZ_tower_job_launch.py
<localhost> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1550596043.35-82061054507257/ /root/.ansible/tmp/ansible-tmp-1550596043.35-82061054507257/AnsiballZ_tower_job_launch.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python2 /root/.ansible/tmp/ansible-tmp-1550596043.35-82061054507257/AnsiballZ_tower_job_launch.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1550596043.35-82061054507257/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_tower_job_launch_payload_VG5rp_/__main__.py", line 129, in main
    result = tower_cli.get_resource(field).get(name=name)
  File "/usr/lib/python2.7/site-packages/tower_cli/models/base.py", line 499, in get
    response = self.read(pk=pk, fail_on_no_results=True, fail_on_multiple_results=True, **kwargs)
  File "/usr/lib/python2.7/site-packages/tower_cli/models/base.py", line 319, in read
    raise exc.NotFound('The requested object could not be found.')

fatal: [phu-b2bsettle-1 -> localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "credential": null,
            "extra_vars": null,
            "inventory": null,
            "job_template": "Jboss restart",
            "job_type": null,
            "limit": "server-1",
            "tags": [
                "stop"
            ]
        }
    },
    "msg": "Unable to launch job, inventory/None was not found: The requested object could not be found."
}
```

**Comment by pravindmali** [comment]
Inventory variables are optional(only need to set when a job template turned on _prompt on launch_ option), its unnecessary to specify when prompt on launch is turned off. PR [54967](https://github.com/ansible/ansible/pull/54967)

**Comment by saito-hideki** [comment]
@dforste can you try to use Ansible v2.8? As @pravindmali pointed out, this problem fixed by PR #54967.
I have confirmed the following playbook worked without errors with Ansible v2.8.1. In v2.8.x, it is not necessary to specify inventory in the playbook task if you already specified it on your job template:

```
- tower_job_launch:
    tower_host: https://tower.example.com/
    tower_username: "{{ secret_tower_user }}"
    tower_password: "{{ secret_tower_password }}"
    tower_verify_ssl: false
    job_template: "Test Job Template"
  register: job
```

**Comment by AlanCoding** [comment]
From the details in the linked Ansible issue, best information indicates this should be resolved in development versions of everything, so closing for now.
